File: solutions/clojure/run-length-encoding/2/src/run_length_encoding.clj
(ns run-length-encoding)

(defn run-length-encode [plaintext]
  (->> plaintext
       (partition-by identity)
       (mapcat (juxt count first))
       (remove #{1})
       (apply str)))

(defn run-length-decode [ciphertext]
  (->> ciphertext
       (re-seq #"(\d*)(\D)")
       (mapcat (fn [[_ n ch]]
                 (repeat (or (parse-long n) 1) ch)))
       (apply str)))

File: solutions/lua/run-length-encoding/2/run-length-encoding.lua
return {
	encode = function(s)
		return s:gsub("((%a)(%a*%2))", function(count, ch) return #count .. ch end)
	end,
	decode = function(s)
		return s:gsub("(%d+)(.)", function(n, ch) return ch:rep(n) end)
	end,
}
